File: backend/src/main/java/com/flowlink/common/error/ApiError.java
package com.flowlink.common.error;

import java.time.Instant;
import java.util.List;

/** 표준 에러 응답 바디 (RFC 7807에 준하는 단순화 형태). */
public record ApiError(
        Instant timestamp,
        int status,
        String error,
        String message,
        String path,
        List<String> details
) {
    public static ApiError of(int status, String error, String message, String path, List<String> details) {
        return new ApiError(Instant.now(), status, error, message, path, details == null ? List.of() : details);
    }
}

File: backend/src/main/kotlin/com/flowlink/common/error/ApiError.kt
package com.flowlink.common.error

import java.time.Instant

/** 표준 에러 응답 바디 (RFC 7807에 준하는 단순화 형태). */
data class ApiError(
    val timestamp: Instant,
    val status: Int,
    val error: String,
    val message: String?,
    val path: String,
    val details: List<String>
) {
    companion object {
        @JvmStatic
        fun of(
            status: Int,
            error: String,
            message: String?,
            path: String,
            details: List<String>?
        ): ApiError =
            ApiError(Instant.now(), status, error, message, path, details ?: emptyList())
    }
}
